lib/manager/actions/user.js

```diff
@@ -242,21 +243,31 @@ export function updateUserData (user: any, userData: any) {
       appMetadata.datatools[dtIndex][key] = userData[key]
     }
     const url = `/api/manager/secure/user/${user.user_id}`
+    let status
     // Make request to server
     return dispatch(secureFetch(url, 'put', {
       user_id: user.user_id,
       data: appMetadata.datatools
     }))
-      .then(response => response.json())
-      .then(updatedProfile => {
+      .then(response => {
+        status = response.status
+        return response.json()
+      })
+      .then(responseJson => {
+        if (status >= 400) {
+          return dispatch(setErrorMessage({
+            message: `Could not update user metadata\n${JSON.stringify(responseJson)}`
+          }))
+        }
         const {profile} = getState().user
         if (!profile) throw new Error('Could not find user profile in state.')
-        const currentUserId = profile.user_id
-        if (updatedProfile.user_id === currentUserId) {
+        if (responseJson.user_id === profile.user_id) {
           // If user being updated matches the logged in user, update their
-          // profile in the application state.
-          dispatch(userProfileUpdated(updatedProfile))
+          // profile in the application state as the change in permissions
+          // may affect what they're allowed to do in the application.
+          dispatch(userProfileUpdated(responseJson))
         }
+        dispatch(fetchUsers())
```

Comment:
I don't think `fetchUsers` should always be called. This function is called in other places besides the user admin module (e.g., `updateTargetForSubscription`). So when a user updates their own subscriptions, we really don't want to be fetching all users. This was the original purpose of the code that updates the logged in user that was removed in a previous commit (and was added back here). Given how much confusion has surrounded this relatively minor code change, I think you should update this method's doc to read:
```
Update application/client ID specific datatools object with provided user data. This is used in the user admin module to update some user's permissions as well as throughout the application to update attributes for the logged in user (e.g., when a user subscribes/watches a feed source or project).

Reply:
Oh wow, yeah, I didn't realize how widely this action was used and how often times it's for updating subscriptions and other non-permissions stuff.